include/counter.h

```diff
@@ -70,6 +70,21 @@ struct counter_alarm_cfg {
  */
 typedef void (*counter_top_callback_t)(struct device *dev, void *user_data);
 
+/** @brief Top value configuration structure.
+ *
+ * @param callback	Callback function. Can be NULL.
+ * @param ticks		Top value.
+ * @param user_data	User data passed to callback function. Not valid if
+ *			callback is NULL.
+ * @param reset		If true, the counter value is being reset too.
+ */
+struct counter_top_cfg {
+	counter_top_callback_t callback;
+	u32_t ticks;
```

Comment:
Shouldn't this be the first field? It's the main one, and not optional. I know, I'm picky.

Reply:
wanted to make if similar as the one above:
```
struct counter_alarm_cfg {
	counter_alarm_callback_t callback;
	u32_t ticks;
	void *user_data;
	bool absolute;
};
```

will flip it.